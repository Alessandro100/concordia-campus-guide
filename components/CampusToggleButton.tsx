import React, { Component } from 'react';
import { TouchableOpacity, View, StyleSheet, Text } from 'react-native';
import Colors from '../constants/Colors';

const styles = StyleSheet.create({
  campusToggle: {
    display: 'flex',
    flexDirection: 'row',
    position: 'absolute',
    top: 110,
    zIndex: 2,
    borderRadius: 10,
    overflow: 'hidden',
  },
  buttonText: {
    color: Colors.white,
    fontWeight: 'bold',
    fontSize: 18,
    textAlign: 'center',
  },
});

class CampusToggleButton extends Component<
  { setMapLocation(latitude: number, longitude: number): void },
  { currentCampusView: String }
> {
  constructor(props) {
    super(props);

    this.state = {
      currentCampusView: 'SGW',
    };
  }

  buttonStyling(campusSelected) {
    const { currentCampusView } = this.state;
    return {
      backgroundColor: campusSelected === currentCampusView ? Colors.primaryColor : Colors.grey,
      opacity: campusSelected === currentCampusView ? 1 : 0.7,
      padding: 7,
      width: 100,
    };
  }

  toggleCampusView(campusSelected) {
    // @ts-ignore
    const { setMapLocation } = this.props;
    const { currentCampusView } = this.state;
    if (currentCampusView === 'SGW' && campusSelected !== 'SGW') {
      this.setState({ currentCampusView: 'Loyola' });
      setMapLocation(45.4582, -73.6405);
    } else if (currentCampusView === 'Loyola' && campusSelected !== 'Loyola') {
      this.setState({ currentCampusView: 'SGW' });
      setMapLocation(45.497406, -73.577102);
    }
  }

  render() {
    return (
      <View style={styles.campusToggle}>
        <TouchableOpacity
          style={this.buttonStyling('SGW')}
          onPress={() => this.toggleCampusView('SGW')}
        >
          <Text id="toggle-sgw" style={styles.buttonText}>
            SGW
          </Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={this.buttonStyling('Loyola')}
          onPress={() => this.toggleCampusView('Loyola')}
        >
          <Text id="toggle-loyola" style={styles.buttonText}>
            Loyola
          </Text>
        </TouchableOpacity>
      </View>
    );
  }
}

export default CampusToggleButton;
