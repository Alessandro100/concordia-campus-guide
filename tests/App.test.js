import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import CampusToggleButton from '../components/CampusToggleButton';

describe('Sample Test Suite', () => {
  test('Sample Test Case', () => {
    expect(true).toEqual(true);
  }),
  test('Sample Snapshot Test', () => {
    const tree = renderer
      .create(<CampusToggleButton />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  }),
  test('Sample Enzyme Test', () => {
    const wrapper = render(<CampusToggleButton />);
    expect(wrapper.find('[testID="toggle-sgw"]').text()).toEqual("SGW");
    expect(wrapper.find('[testID="toggle-loyola"]').text()).toEqual("Loyola");
  });
  test('Sample Enzyme Test Current Location button', () => {
    const wrapperMount = mount(<App/>);
    expect(wrapperMount.find(TouchableHighlight)).toHaveLength(1);
    expect(wrapperMount.find(Image)).toHaveLength(1);
  })
});