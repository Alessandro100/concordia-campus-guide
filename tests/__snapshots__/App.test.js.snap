// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample Test Suite Sample Snapshot Test 1`] = `
<View
  style={
    Object {
      "borderRadius": 10,
      "display": "flex",
      "flexDirection": "row",
      "overflow": "hidden",
      "position": "absolute",
      "top": 110,
      "zIndex": 2,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#932439",
        "opacity": 1,
        "padding": 7,
        "width": 100,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
      testID="toggle-sgw"
    >
      SGW
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#746D75",
        "opacity": 0.7,
        "padding": 7,
        "width": 100,
      }
    }
    testID="toggle-loyola-button"
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
      testID="toggle-loyola"
    >
      Loyola
    </Text>
  </View>
</View>
`;
